### PR TITLE
Userdetails form added and also verification check screen

### PR DESCRIPTION
**Pull request by kshitijalwadhi**

### CODE REVIEW

File: lib/screens/authenticate/verified_email_check.dart [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        var user = await FirebaseAuth.instance.currentUser();
+        if (user.isEmailVerified) {
+          setState(() {
+            verified = user.isEmailVerified;
+          });
+          timer.cancel();
+        }
+      });
+    });
+    _timer = _timer;
```
Comment (arpit-saxena) [inline lib/screens/authenticate/verified_email_check.dart:52]: What does this line achieve?

Comment (arpit-saxena) [reply lib/screens/authenticate/verified_email_check.dart:52]: Maybe mention in the comments what you are trying to do.

File: lib/services/auth.dart [hunk 2]
```diff
@@ -22,12 +23,26 @@ class AuthService {
     }
   }
 
+  Future<bool> checkVerification(FirebaseUser user) async {
+    if (user.isEmailVerified) {
+      return true;
+    } else {
+      return false;
```
Comment (arpit-saxena) [inline lib/services/auth.dart:30]: Simply return `user.isEmailVerified`

File: lib/services/auth.dart [hunk 3]
```diff
@@ -22,12 +23,26 @@ class AuthService {
...
+    }
+  }
+
   // sign up with email pass
 
-  Future<void> registerWithEmailAndPassword(
-      String email, String password) async {
-    var result = await _auth.createUserWithEmailAndPassword(
+  Future<void> registerWithEmailAndPassword(String email, String password,
+      String name, String mobilenum, String hostel, String sex) async {
```
Comment (arpit-saxena) [inline lib/services/auth.dart:37]: This has too many arguments for it to use positional parameters. Convert them to named parameters. Change the call appropriately

File: lib/services/database.dart [hunk 4]
```diff
@@ -8,13 +9,37 @@ class DatabaseService {
   final CollectionReference userDetails =
       Firestore.instance.collection('userdetails');
 
-  Future initializeUserData(
-      String name, int mobileNumber, int hostel, int sex) async {
+  Future enterUserData(
+      String name, String mobileNumber, String hostel, String sex) async {
```
Comment (arpit-saxena) [inline lib/services/database.dart:13]: Use named parameters here too

File: lib/screens/authenticate/verified_email_check.dart (hunk 1 above)
Comment (kshitijalwadhi) [reply lib/screens/authenticate/verified_email_check.dart:52]: Flutter analyze was showing an error that _timer isn't being used anywhere, idk why. Hence this line to bypass that :p

File: lib/services/auth.dart (hunk 2 above)
Comment (kshitijalwadhi) [reply lib/services/auth.dart:30]: changed

File: lib/screens/authenticate/register.dart [hunk 5]
```diff
@@ -80,31 +208,36 @@ class _RegisterState extends State<Register> {
                           if (_formKey.currentState.validate()) {
                             setState(() => loading = true);
                             try {
-                              await _auth.registerWithEmailAndPassword(
-                                  email, password);
+                              setState(() {
+                                email = email.trim();
+                              });
```
Comment (tmibvishal) [inline lib/screens/authenticate/register.dart:213]: Why using setState only to trip an email. Just trim it while sending a request. setState is expensive as it changes resources on screen

File: lib/screens/authenticate/sign_in.dart [hunk 6]
```diff
@@ -60,12 +68,26 @@ class _SignInState extends State<SignIn> {
...
                       TextFormField(
-                        decoration:
-                            textInputDecoration.copyWith(hintText: 'Password'),
+                        decoration: textInputDecoration.copyWith(
+                          hintText: 'Password',
+                          suffixIcon: IconButton(
+                            icon: Icon(
+                              passwordHide
+                                  ? Icons.visibility_off
+                                  : Icons.visibility,
```
Comment (tmibvishal) [inline lib/screens/authenticate/sign_in.dart:77]: Although not a big issue, clicking on IconButton ripples hide behind the textbox. It should be in front.

File: lib/services/auth.dart (hunk 3 above)
Comment (kshitijalwadhi) [reply lib/services/auth.dart]: fixed

File: lib/services/database.dart (hunk 4 above)
Comment (kshitijalwadhi) [reply lib/services/database.dart]: fixed

File: lib/screens/authenticate/register.dart (hunk 5 above)
Comment (kshitijalwadhi) [reply lib/screens/authenticate/register.dart:213]: fixed

File: lib/screens/authenticate/verified_email_check.dart (hunk 1 above)
Comment (ankurshaswat) [reply lib/screens/authenticate/verified_email_check.dart:52]: If you're not using the returned value you could just remove the _timer =
Just treat it as no return statement.

**Comment by ankurshaswat** [comment]
Run tests locally before pushing.

File: lib/screens/authenticate/verified_email_check.dart (hunk 1 above)
Comment (kshitijalwadhi) [reply lib/screens/authenticate/verified_email_check.dart:52]: fixed, will push it along next set of feature additions.

**Pull request merged**